default/tests/shared/test_annotation_new.py

```diff
@@ -0,0 +1,273 @@
+from methods.regular.regular_api import *
+from default.tests.test_utils import testing_setup
+from shared.annotation import Annotation_Update
+from unittest.mock import patch, MagicMock, Mock, call
+from unittest import skip
+
+class TestAnnotationUpdate(testing_setup.DiffgramBaseTestCase):
+
+    def setUp(self):
+        super(TestAnnotationUpdate, self).setUp()
+        self.mock_session = MagicMock()
+        self.mock_project = MagicMock()
+        self.mock_member = MagicMock()
+        self.mock_file = MagicMock()
+        self.instance = Annotation_Update(session=self.mock_session, file=self.mock_file, member=self.mock_member, project=self.mock_project)
+  
+    # TODO: How can we clean this up / re-use all the setup?
+
+    @patch('shared.annotation.regular_log.default')
+    @patch('shared.annotation.Project.get_by_id')
+    @patch('shared.annotation.get_member')
+    @patch('shared.annotation.Annotation_Update.get_allowed_label_file_ids')
+    @patch('shared.annotation.Annotation_Update.init_video_input')
+    @patch('shared.annotation.Annotation_Update.task_update')
+    @patch('shared.annotation.Annotation_Update.init_file')
+    @patch('shared.annotation.Annotation_Update.init_existing_instances')
+    @patch('shared.annotation.Annotation_Update.refresh_instance_count')
+    def test_post_init_main_functions_called(self, mock_refresh_instance_count, mock_init_existing_instances, mock_init_file, mock_task_update, mock_init_video_input, mock_get_allowed_label_file_ids, mock_get_member, mock_Project_get_by_id, mock_default_log):
+        # Arrange
+        # func_calls = Mock()
+        # func_calls.m1, func_calls.m2, func_calls.m3, func_calls.m4, func_calls.m5, func_calls.m6 = mock_get_allowed_label_file_ids, mock_init_video_input, mock_task_update, mock_init_file, mock_init_existing_instances, mock_refresh_instance_count
+
+        # Act
+        instance = Annotation_Update(session=self.mock_session, file=self.mock_file, member=self.mock_member, project=self.mock_project)
+
+        # Assert
+        self.assertEqual(instance.log, mock_default_log.return_value) # Log was initialized
+        mock_Project_get_by_id.assert_not_called() # Project was not fetched
+        mock_get_member.assert_not_called() # Member was not fetched
+
+        # TODO: This isnt working, got it from here https://stackoverflow.com/questions/32463321/how-to-assert-method-call-order-with-python-mock
+        # func_calls.assert_has_calls([call.m1(), call.m2(), call.m3(), call.m4(), call.m5(), call.m6()])
```

Comment:
Wanted to add an assert to ensure we call each of these functions in the correct order, but couldnt get this strategy to work. Will revisit later if we still depend on the ordering.